### PR TITLE
Unfocus text fields when canvas clicked and don't take keystrokes in canvas when text fields are focused

### PR DESCRIPTION
**Pull request by fsih**
Fix some issues where you can edit text and the paint editor at the same time.

### CODE REVIEW

File: src/helper/selection-tools/reshape-tool.js [hunk 1]
```diff
@@ -232,6 +232,11 @@ class ReshapeTool extends paper.Tool {
         this.active = false;
     }
     handleKeyDown (event) {
+        if (event.event.target instanceof HTMLInputElement) {
+            // Ignore nudge if a text input field is focused
+            return;
+        }
```
Comment (rschamp) [inline src/helper/selection-tools/reshape-tool.js:238]: Is this here intentionally? Looks unrelated. Probably fine to leave it though.

File: src/containers/paint-editor.jsx [hunk 2]
```diff
@@ -135,6 +141,10 @@ class PaintEditor extends React.Component {
         this.canvas = canvas;
     }
     onMouseDown () {
+        if (document.activeElement instanceof HTMLInputElement) {
```
Comment (rschamp) [inline src/containers/paint-editor.jsx]: To keep this contained to this specific issue, would you also check that the mousedown event is on the paint editor canvas?

**Review by rschamp — CHANGES_REQUESTED** [review]

File: src/helper/selection-tools/reshape-tool.js (hunk 1 above)
Comment (fsih) [reply src/helper/selection-tools/reshape-tool.js:238]: This is vaguely related. Currently, if you are on the reshape tool and you have something selected, and you also have a textbox focused, if you press the arrow keys, both the selected thing and the cursor position will move. This return makes it so that only the cursor position moves. I will update the description.

File: src/containers/paint-editor.jsx (hunk 2 above)
Comment (fsih) [reply src/containers/paint-editor.jsx]: Oops, thanks for catching

**Review by rschamp — APPROVED** [review]
